lake/passes/passes.py

```diff
@@ -47,3 +48,49 @@ def visit(self, node):
 
     v = ConfigRegLiftVisitor()
     v.visit_root(generator)
+
+
+# Inputs:
+#   - use_sram_stub: if stub is not being used, we are using external provided
+#       sram macro and should replace port names accordingly
+#   - ports: the port names for the sram macro, provided in alphabetical order
+#       for replacement
+#   - testing: boolean indicating whether we are generating verilog (in which
+#       case the generator is automatically provided) or whether we are testing
+#       (and need to specify the generator explicitly)
+#   - generator: explicit specification of generator if needed
+def change_sram_port_names(use_sram_stub, ports, testing, generator):
+
+    def change_sram_port_names_wrapper(generator):
+
+        class SRAMPortNames(IRVisitor):
+            def __init__(self, use_sram_stub, ports):
+                IRVisitor.__init__(self)
+                self.use_sram_stub = use_sram_stub
+                self.ports = ports
+
+            def visit(self, node):
+                if isinstance(node, Port):
+                    if not(len(node.get_attributes()) <= 0 or self.use_sram_stub):
+                        for i in range(len(node.get_attributes())):
+                            if (isinstance(node.get_attributes()[i].get(), SRAMPortAttr)):
+                                if node.name == "sram_addr":
+                                    node.name = self.ports[0]
```

Comment:
Why can't you use `node.name = self.addr_port_name`?

Reply:
Oh it does the same thing, but your way is more readable, so I will change it.